pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,343 @@
+	ctx, log := logger.GetNewContextWithLogger()
+	// Fetch the CnsFileAccessConfig instance
+	instance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Infof("CnsFileAccessConfig resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Error reading the CnsFileAccessConfig with name: %q on namespace: %q. Err: %+v",
+			request.Name, request.Namespace, err)
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	backOffDurationMapMutex.Lock()
+	var timeout time.Duration
+	if _, exists := backOffDuration[instance.Name]; !exists {
+		backOffDuration[instance.Name] = time.Second
+	}
+	timeout = backOffDuration[instance.Name]
+	backOffDurationMapMutex.Unlock()
+
+	// If the CnsFileAccessConfig instance is already successful,
+	// and not deleted by the user, remove the instance from the queue.
+	if instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		return reconcile.Result{}, nil
+	}
+
+	log.Infof("Reconciling CnsFileAccessConfig with instance: %q from namespace: %q. timeout %q seconds", instance.Name, instance.Namespace, timeout)
+	if !instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Get the virtualmachine instance
+		vm, err := getVirtualMachine(ctx, instance.Spec.VMName, instance.Namespace)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get virtualmachine instance for the VM with name: %q. Error: %+v", instance.Spec.VMName, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found virtualMachine instance for VM with name: %q: +%v", instance.Spec.VMName, vm)
+
+		// Set ownerRef on CnsFileAccessConfig instance (in-memory) to VM instance
+		setInstanceOwnerRef(instance, instance.Spec.VMName, vm.UID)
+
+		cnsFinalizerExists := false
+		// Check if finalizer already exists.
+		for _, finalizer := range instance.Finalizers {
+			if finalizer == cnsoperatortypes.CNSFinalizer {
+				cnsFinalizerExists = true
+				break
+			}
+		}
+		// Update finalizer and ownerRef on CnsFileAccessConfig instance
+		if !cnsFinalizerExists {
+			// Add finalizer.
+			instance.Finalizers = append(instance.Finalizers, cnsoperatortypes.CNSFinalizer)
+			err = updateCnsFileAccessConfig(ctx, r.client, instance)
+			if err != nil {
+				msg := fmt.Sprintf("failed to update CnsFileAccessConfig instance: %q on namespace: %q. Error: %+v",
+					instance.Name, instance.Namespace, err)
+				recordEvent(ctx, r, instance, v1.EventTypeWarning, msg)
```

Comment:
Why do we use `recordEvent` in some places and `setInstanceSuccess/Err` in others?

Reply:
`recordEvent` : records the event and also sets the backOffDuration for the instance
It is invoked explicitly at 2 places: One(1) is for recording a Get failure and one(2) while updating finalizers.

Otherwise `recordEvent` is called inside setInstanceSuccess/setInstanceErr.
And the two explicit invocations have nothing to do with instance.Spec or instance.Status. 
I do not see anything incorrect here. PTAL